py/dml/codegen.py

```diff
@@ -1260,10 +1260,19 @@ def try_resolve_len(site, lh):
 def expr_member(tree, location, scope):
     [lh, op, rh] = tree.args
     lh = codegen_expression_maybe_nonvalue(lh, location, scope)
-    if not tree.site.dml_version() == (1, 2) and op == '.' and rh == 'len':
-        member = try_resolve_len(tree.site, lh)
-        if member:
-            return member
+    if op == '.':
+        if not tree.site.dml_version() == (1, 2) and rh == 'len':
+            member = try_resolve_len(tree.site, lh)
+            if member:
+                return member
+        elif isinstance(lh, TemplatesRef):
+            return mkTemplatesSubRef(tree.site, lh, rh)
+        elif isinstance(lh, TemplatesSubRef):
+            return mkTemplateQualifiedMethodRef(tree.site, lh, rh)
+        elif isinstance(lh, TraitTemplatesRef):
+            return mkTraitTemplatesSubRef(tree.site, lh, rh)
+        elif isinstance(lh, TraitTemplatesSubRef):
+            return mkTraitTemplateQualifiedMethodRef(tree.site, lh, rh)
```

Comment:
I think other similar things are resolved in mkSubRef.

Looking at mkSubRef raises another corner:
```
template x {
  method x() {}
}
template y { is x; }
group g is y;
...
local y y = cast(g, y);
local y *yp = &y;
// should we choke on `->`?
local x x = yp->template.x;
```

Reply:
> I think other similar things are resolved in mkSubRef
Actually no. `mkSubRef` resolves that which is type-driven, while `expr_member` resolves that which is  expression-class-driven. That said there is no good reason for that discrepency, so everything that's in `expr_member` may as well be moved to `mkSubRef`.
```
local x x = yp->templates.x;
```
We may as well permit it. Easy with the move into `mkSubRef`.

Reply:
Actually `local x x = yp->templates.x;` is already permitted; that is actually resolved via `mkSubRef`. Thus the only question is whether the non-type-driven components should be part of `mkSubRef`, and I'm thinking "why not?"

Reply:
I've made the move to `mkSubRef`. Resolve this if you're satisfied.